### ISSUE TITLE
Add support for websocket (ws and wss) in bootstrap node

### ISSUE THREAD

**Issue by iphydf** [TokTok/c-toxcore#2584]
This is useful for supporting webtox and later to get around application layer firewalls like the ones in airports that only allow http connections.

We can look into mbed TLS as a library for crypto needed for https/wss. For webtox, that's a requirement. For native tox, we don't need encrypted websockets since we'll be sending tox crypto over it anyway.

**Comment by emdee-is** [comment]
Does websocket (ws and wss) in bootstrap node work over Tor?

If not, and I suspect not, then it raises the question of priority planning in the project: does the project want Tox in Tor more than it wants websockets?

There is a prominent wiki that would be good for project planning on [wiki.tox.chat](https://wiki.tox.chat/doku.php) that I think @nurupo maintains, but you would need people to use it, and to prioritize things. (I made an accout there and applied for write permissions by never heard back as I wamted to add some of the pages from [my wiki](https://git.plastiras.org/emdee/tox_profile/wiki). There's a lot of outdated information there that needs cleaning up: the people interested in the project must realize that it is the public face of Tox.

The reason I suggest this is because I'm seeing a lot of "rewrite-from-scratch" which can be a planning failure if it's not been thought out (think Netscape) and for that you need discussion in something like a wiki. You can't use NGC for that sort of thing, and you have no history of when you're not connected.

**Comment by iphydf** [comment]
Yes, it works over tor. Websockets are TCP.

**Comment by emdee-is** [comment]
> Yes, it works over tor. Websockets are TCP.

Yes I know they're TCP but I don't think websockets are easy to run over Tor. I'll check to see, but I think the Tor browser disables them, perhaps because of IP leakage.

To see what it takes to use them over Tor see: https://meejah.ca/blog/end-to-end-encrypted-rpc-over-tor and WAMP https://blog.torproject.org/secure-messaging-onion-services-how/ but I don't think any of that works these days, although I'm no expert at it.

No feeback on the idea of project planning and prioritization?

**Comment by emdee-is** [comment]
I'll take that as No to the idea of project planning and prioritization.

If the project changes it mind and decides to proritize the work, here is my list of suggestions :

- support Python https://github.com/TokTok/py-toxcore-c/issues/78#issuecomment-1920417914
- support Corporate proxies https://github.com/TokTok/c-toxcore/issues/1682
- test Tor proxies https://github.com/TokTok/c-toxcore/issues/2469#issuecomment-1858986719
- test Corporate proxies https://github.com/TokTok/c-toxcore/issues/1682
- support Tox in Tor
  -   blockers: https://github.com/TokTok/c-toxcore/issues/2467
- actively support an IRC channel for supporting new users https://github.com/TokTok/c-toxcore/issues/2463
- adopt one of the abandonned GUI clients into TokTok so that there's at least one supported GUI client,
-  integrate the testing of the supported GUI client.
- bring the https://wiki.tox.chat up to date and mark the abandonned clients - it is the visible face of Tox.
- start project planning and prioritization, perhaps using the wiki
